app/Static/main-7.js

```diff
@@ -321,6 +321,33 @@ function Disab_Form() {
   $("a.btn").addClass("disabled");
 }
 
+$(document).ready(function(){
+  $('.favorite-icon').on('click', async function(e){
+    e.preventDefault();
+    const $icon     = $(this);
+    const productId = $icon.data('product-id');
+
+    const temp = $icon.attr('src');
+    $icon.attr('src', $icon.data('other-src'));
+    $icon.data('other-src', temp);
+
+    const oDataReq = { productId };
+
+    const oOptsFetch = {
+      ...OptsFetchCSRF(),
+      method: "POST",
+      body: JSON.stringify(oDataReq),
+    };
+    const oResp = await fetch("/toggle-favorite", oOptsFetch);
+    if (!oResp.ok) {
+      const temp = $icon.attr('src');
+      $icon.attr('src', $icon.data('other-src'));
```

Comment:
this totally works. If you wanted to make this work with preloaded images, you could [use a css class](https://stackoverflow.com/a/11605210/3430807) then [toggle the class](https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/toggle#toggling_a_class_on_click)